File: Java/exercicio3.java
class ContaBancaria {
    private double saldo;
    private String titular;

    public ContaBancaria(String titular) {
        this.titular = titular;
        this.saldo = 0;
    }

    public void depositar(double valor) {
        this.saldo += valor;
    }

    public void sacar(double valor) throws Exception {
        if (valor > saldo) {
            throw new Exception("Saldo insuficiente");
        }
        this.saldo -= valor;
    }

    public double obterSaldo() {
        return saldo;
    }
}

public class Main {
    public static void main(String[] args) {
        try {
            ContaBancaria conta = new ContaBancaria("João");
            conta.depositar(100);
            System.out.println(conta.obterSaldo());
            conta.sacar(50);
            System.out.println(conta.obterSaldo());
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }
}

File: Python/exercicio3.py
class ContaBancaria:
    def __init__(self, titular):
        self._saldo = 0
        self.titular = titular

    def depositar(self, valor):
        self._saldo += valor

    def sacar(self, valor):
        if valor > self._saldo:
            raise Exception("Saldo insuficiente")
        self._saldo -= valor

    def obter_saldo(self):
        return self._saldo

conta = ContaBancaria("João")
conta.depositar(100)
print(conta.obter_saldo())
conta.sacar(50)
print(conta.obter_saldo())